create/run: remove default --stop-signal

thaJeztah opened the pull request:
relates to https://github.com/moby/moby/pull/42717

The DefaultStopSignal const has been deprecated, because the daemon already
handles a default value. The current code did not actually send the default
value unless the flag was set, which also made the flag description incorrect,
because in that case, the _daemon's_ default would be used, which could
potentially be different as was specified here.

This patch removes the default value from the flag, leaving it to the daemon
to set a default.

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

```
Remove default value from `--stop-signal` flag, as it may not reflect the actual default used by the daemon.
```


**- A picture of a cute animal (not mandatory but encouraged)**

tianon commented on man/docker-run.1.md:
```diff
@@ -621,7 +621,7 @@ incompatible with any restart policy other than `none`.
 
-**--stop-signal**=*SIGTERM*
+**--stop-signal**=""
   Signal to stop a container. Default is SIGTERM.
```
Should this maybe note the fact that the default can be defined by the image or something? (`STOPSIGNAL` in `Dockerfile`)

```suggestion
  Signal to stop a container. Default is defined by `STOPSIGNAL` in the image or `SIGTERM` if not defined.
```

thaJeztah replied:
Ah, yes, makes sense to add, let me add some of that; thanks!

thaJeztah commented:
I did some further rewriting of the docs, and moved a large part to https://github.com/docker/cli/pull/3261 (included in this PR; first two commits are from there)

thaJeztah commented on cli/command/container/kill.go:
```diff
@@ -32,7 +32,7 @@ func NewKillCommand(dockerCli command.Cli) *cobra.Command {
 
 	flags := cmd.Flags()
-	flags.StringVarP(&opts.signal, "signal", "s", "KILL", "Signal to send to the container")
+	flags.StringVarP(&opts.signal, "signal", "s", "SIGKILL", "Signal to send to the container")
```
I'm considering to remove this default from the client as well, but keeping that for a follow-up

thaJeztah commented:
Rebased, to removed the changes from https://github.com/docker/cli/pull/3261 from this PR, and moved the `KILL -> SIGKILL` changes separate as well, as it's not directly related https://github.com/docker/cli/pull/3271

thaJeztah commented:
@tianon @silvin-lubecki PTAL

tianon reviewed (approved):
:eyes:

thaJeztah commented:
@silvin-lubecki PTAL 🤗

thaJeztah commented:
@silvin-lubecki @tonistiigi PTAL 🤗

silvin-lubecki reviewed (approved):
LGTM
